### ISSUE TITLE
False positive deprecated-method with alias to non-deprecated method

### ISSUE THREAD

**Issue by kjamieson** [PyCQA/pylint#5886]
### Bug description

With pylint 2.12.2 and python 3.8 the following code:

```python
import time

def test(clock = time.time):
    clock()
```

Produces a false-positive `deprecated-method` warning:

```
test.py:4:4: W1505: Using deprecated method clock() (deprecated-method)
```

I think this is related to the deprecated alias detection added in https://github.com/PyCQA/pylint/issues/4618 as the issue does not reproduce with that change reverted.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:4:4: W1505: Using deprecated method clock() (deprecated-method)
```

### Expected behavior

No warning.

### Pylint version

```shell
2.12.2
```

### OS / Environment

Ubuntu 18.04

**Comment by Pierre-Sassoulas** [comment]
I can't reproduce on 2.15.0-dev0, I suppose it was fixed in https://github.com/PyCQA/pylint/pull/4619

**Comment by Pierre-Sassoulas** [comment]
Reopening in order to close with a regression test.

**Comment by clavedeluna** [comment]
I'm actually able to reproduce this with pylint 2.15.4, so a fix + regression test is needed.
